### PR TITLE
fix elasticsearch incompatibility problem

### PR DESCRIPTION
**Pull request by miiu96**

### CODE REVIEW

File: core/indexer/elasticsearchDatabase_test.go [hunk 1]
```diff
@@ -327,3 +327,34 @@ func TestElasticsearch_saveRoundInfoRequestError(t *testing.T) {
 
 	require.True(t, strings.Contains(output.String(), localError.Error()))
 }
+
+// Commented test works only if you have an elastic search server with address "http://localhost:9200"
```
Comment (ccorcoveanu) [inline core/indexer/elasticsearchDatabase_test.go:331]: Remove commented code if not needed

File: core/indexer/elasticsearchDatabase.go [hunk 2]
```diff
@@ -278,14 +278,17 @@ func (esd *elasticSearchDatabase) SaveMiniblocks(header data.HeaderHandler, body
 	}
 
 	buffInsert, buffUpdate := serializeBulkMiniBlocks(header.GetShardID(), miniblocks)
-	err := esd.dbWriter.DoBulkRequest(&buffInsert, miniblocksIndex)
-	if err != nil {
-		log.Warn("indexing bulk of miniblock insert ", "error", err.Error())
+	if buffInsert.Len() != 0 {
```
Comment (ccorcoveanu) [inline core/indexer/elasticsearchDatabase.go:281]: In the end we are going to leave 2 separate requests for insert and update? I remember that in the end we realised that it was good in one request also. But not an issue if we do it separately.

File: core/indexer/elasticsearchDatabase_test.go (hunk 1 above)
Comment (bogdan-rosianu) [reply core/indexer/elasticsearchDatabase_test.go:331]: agree, remove this test

File: core/indexer/elasticsearchDatabase.go [hunk 3]
```diff
@@ -278,14 +278,17 @@ func (esd *elasticSearchDatabase) SaveMiniblocks(header data.HeaderHandler, body
...
+		err := esd.dbWriter.DoBulkRequest(&buffInsert, miniblocksIndex)
+		if err != nil {
+			log.Warn("indexing bulk of miniblock insert ", "error", err.Error())
+		}
 	}
-
-	err = esd.dbWriter.DoBulkRequest(&buffUpdate, miniblocksIndex)
-	if err != nil {
-		log.Warn("indexing bulk of miniblock update ", "error", err.Error())
+	if buffUpdate.Len() != 0 {
```
Comment (bogdan-rosianu) [inline core/indexer/elasticsearchDatabase.go:287]: `> 0 ` ? also above

File: core/indexer/elasticsearchDatabase.go [hunk 4]
```diff
@@ -278,14 +278,17 @@ func (esd *elasticSearchDatabase) SaveMiniblocks(header data.HeaderHandler, body
...
+		err := esd.dbWriter.DoBulkRequest(&buffUpdate, miniblocksIndex)
+		if err != nil {
+			log.Warn("indexing bulk of miniblock update ", "error", err.Error())
```
Comment (bogdan-rosianu) [inline core/indexer/elasticsearchDatabase.go:290]: miniblocks* also above

Comment (miiu96) [reply core/indexer/elasticsearchDatabase.go:290]: done.

File: core/indexer/elasticsearchDatabase.go (hunk 3 above)
Comment (miiu96) [reply core/indexer/elasticsearchDatabase.go:287]: done.

File: core/indexer/elasticsearchDatabase_test.go (hunk 1 above)
Comment (miiu96) [reply core/indexer/elasticsearchDatabase_test.go:331]: i will use t.skip(...) for this test

File: core/indexer/elasticsearchDatabase.go (hunk 2 above)
Comment (miiu96) [reply core/indexer/elasticsearchDatabase.go:281]: You right is good to do only one request when we index miniblocks. Fixed .

File: core/indexer/elasticsearchDatabase.go [hunk 5]
```diff
@@ -277,18 +277,10 @@ func (esd *elasticSearchDatabase) SaveMiniblocks(header data.HeaderHandler, body
...
-			log.Warn("indexing bulk of miniblock insert ", "error", err.Error())
-		}
-	}
-	if buffUpdate.Len() != 0 {
-		err := esd.dbWriter.DoBulkRequest(&buffUpdate, miniblocksIndex)
-		if err != nil {
-			log.Warn("indexing bulk of miniblock update ", "error", err.Error())
-		}
+	buff := serializeBulkMiniBlocks(header.GetShardID(), miniblocks)
+	err := esd.dbWriter.DoBulkRequest(&buff, miniblocksIndex)
```
Comment (ccorcoveanu) [inline core/indexer/elasticsearchDatabase.go:281]: The condition should stay as a precaution, do not call es server if `len(buff)` is 0

Comment (miiu96) [reply core/indexer/elasticsearchDatabase.go:281]: Not need anymore because few lines above is verified if len(miniblocks) == 0 and return

**Pull request merged**